lua/tests/gmod/classes/file/EndOfFile.lua

```diff
@@ -0,0 +1,24 @@
+return {
+    groupName = "File:EndOfFile",
+
+    cases = {
+        {
+            name = "Exists on the File meta table",
+            func = function()
+                local meta = FindMetaTable( "File" )
+                expect( meta.EndOfFile ).to.beA( "function" )
+            end
+        },
+
+        {
+            name = "Returns the right value",
+            func = function()
+                local a = GetTestFile( "Write" )
+
+                a:Read()
+
+                expect( a:EndOfFile() ).to.equal( true )
+            end
+        },
```

Comment:
Also a negative case?

```suggestion
        {
            name = "Returns true when at the end of the file",
            func = function()
                local a = GetTestFile( "Write" )

                a:Read()

                expect( a:EndOfFile() ).to.beTrue()
            end
        },
        {
            name = "Returns false when not at the end of the file",
            func = function()
                local a = GetTestFile( "Write" )

                expect( a:EndOfFile() ).to.beFalse()
            end
        },
```